### ISSUE TITLE
Automatically set the date/time format based on the currently set language

### ISSUE THREAD

**Issue by romainpoirier** [verbb/formie#520]
**What are you trying to do?**
Currently, when setting up a Date/Time field we can customize the Date Format.
This is a nice feature, but sometimes we have to use several formats based on the current website locale.
This would be nice to have to ability to automatically set the format based on the currently set language (from `craft.app.language`).

**Comment by engram-design** [comment]
Just looking into this one, and it's starting to get complicated pretty quick, and really needs the ability to be able to define these fields so they're site-aware, but localised forms are way, way in the future.

The reason being _what_ format should be used? Even for different languages, there's still multiple ways you might want to format a date (or time) for your users. In (Australian) English for example, all these are valid:

```
Monday 18th of October, 2021
2021/10/18
18/10/2021
18-10-2021
October 18, 2021
```

And picking which ones really comes down to what you prefer, what makes sense to your end-users, client needs, etc. You've got this ability in the field settings for a date field, but you **also** need another layer to set that up for each site.

So instead of enforcing a specific format depending on the site you're on, I've added events for you to control this behaviour, which should help.

```php
use verbb\formie\events\ModifyDateTimeFormatEvent;
use verbb\formie\fields\formfields\Date;
use yii\base\Event;

Event::on(Date::class, Date::EVENT_MODIFY_DATE_FORMAT, function(ModifyDateTimeFormatEvent $event) {
    // Check the current site, change as appropriate
    $event->dateFormat = 'Y-m-d';
});
```

Added for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.20",
  "...": "..."
}
```

Then run `composer update`.
